test/db-backup-restore.spec.ts

```diff
@@ -0,0 +1,196 @@
+import path from 'path'
+import fs from 'fs'
+import config from 'config'
+import chai from 'chai'
+import sinonChai from 'sinon-chai'
+import { Eth } from 'web3-eth'
+import { Substitute, Arg } from '@fluffy-spoon/substitute'
+
+import DbBackUpJob from '../src/db-backup'
+import { blockMock, rmDir, sleep } from './utils'
+import { NEW_BLOCK_EVENT_NAME } from '../src/blockchain/new-block-emitters'
+import { DbBackUpConfig } from '../src/definitions'
+
+chai.use(sinonChai)
+const expect = chai.expect
+
+describe('DB back-up/restore', function () {
+  const configBackUp = { db: config.get('db'), dbBackUp: config.get('dbBackUp') }
+
+  describe('Restore', () => {
+    before(() => fs.writeFileSync(config.get<string>('db'), 'Initial DB'))
+    after(() => fs.unlinkSync(config.get<string>('db')))
+    afterEach(() => rmDir(config.get<DbBackUpConfig>('dbBackUp').path))
+
+    it('should throw if not enough backups', async () => {
+      const eth = Substitute.for<Eth>()
+      const backupJob = new DbBackUpJob(eth)
+
+      expect(fs.readdirSync(config.get<DbBackUpConfig>('dbBackUp').path).length).to.be.eql(0)
+      await expect(backupJob.restoreDb()).to.eventually.be.rejectedWith(
+        Error,
+        'Should be two backups to be able to restore'
+      )
+    })
+    it('should throw if backup block hash is not exist after reorg', async () => {
```

Comment:
Can you please add test-case for when you get a different hash (eq. the reorg is bigger then the backup)?